src/WidgetBase.ts

```diff
@@ -385,6 +391,13 @@ export class WidgetBase<P extends WidgetProperties> extends Evented implements W
 			const { children, properties = {} } = dNode;
 			const { key } = properties;
 
+			const propagatedPropertyNames = this.getDecorator('propagateProperty');
```

Comment:
Do you think this would be better as an `@afterRender` that utilises the decorator from `d.ts`? so runs after each render rather than in dNodeToVNode?